### ISSUE TITLE
__getitem__ uses label-based pandas indexing instead of positional — KeyError with string obs_names

### ISSUE THREAD

**Issue by GravityBoi** [ratschlab/aestetik#7]
CustomDataset.__getitem__ accesses cluster labels using self.label_transcriptomics[idx] and self.label_morphology[idx], where idx is a positional integer from the DataLoader. Pandas
  Series.__getitem__ with an integer uses label-based indexing, not positional.

  This fails in two ways:

  1. String obs_names (standard AnnData with barcode strings): raises FutureWarning today and will raise KeyError in pandas >= 3.0.
  2. After _compute_train_idx (when train_size is set): train_test_split produces a non-contiguous index like [0, 3, 7, 42, ...]. Then series[1] raises KeyError because label 1 is not in the
   index.

  # dataloader.py line 142 — current (broken)
  self.label_transcriptomics[idx], self.label_morphology[idx]

  # fix
  self.label_transcriptomics.iloc[idx], self.label_morphology.iloc[idx]
